manifests/config/entry.pp

```diff
@@ -30,7 +30,7 @@
   # they make sure that '1_main ' is ordered before '1_main_*'
   ensure_resource('concat::fragment', "puppet.conf_${section}", {
       target  => "${puppet::dir}/puppet.conf",
-      content => "\n[${section}]",
+      content => "\n\n[${section}]",
```

Comment:
The only downside of this is that you end up with 2 empty lines at the start of the file. I'm not sure how to best deal with that.